tests/metrics/de/test_reference_based_text_quality.py

```diff
@@ -0,0 +1,155 @@
+import os
+from unittest.mock import Mock, patch
+
+import pytest
+from openai.types import CreateEmbeddingResponse
+
+from langcheck.metrics.de import rouge1, rouge2, rougeL, semantic_similarity
+from tests.utils import is_close
+
+################################################################################
+# Tests
+################################################################################
+
+
+@pytest.mark.parametrize(
+    "generated_outputs,reference_outputs",
+    [
+        ("Die Katze saß auf der Matte.", "Die Katze saß auf der Matte."),
+        (["Die Katze saß auf der Matte."], ["Die Katze saß auf der Matte."]),
+    ],
+)
+def test_semantic_similarity_identical(generated_outputs, reference_outputs):
+    metric_value = semantic_similarity(generated_outputs,
+                                       reference_outputs,
+                                       model_type="local")
+    assert 0.99 <= metric_value <= 1
+
+
+@pytest.mark.parametrize(
+    "generated_outputs,reference_outputs",
+    [
+        ("Die KATZE saß auf der MATTE.", "Die Katze saß auf der Matte."),
+        (["Die KATZE saß auf der MATTE."], ["Die Katze saß auf der Matte."]),
+    ],
+)
+def test_semantic_similarity_case_sensitivity(generated_outputs,
+                                              reference_outputs):
+    # nb: the German model is case-sensitive!
+    metric_value = semantic_similarity(generated_outputs,
+                                       reference_outputs,
+                                       model_type="local")
+    assert 0.6 <= metric_value <= 0.7
```

Comment:
interesting! is this an expected result? naively I'd expect two sentences where only the case is different to have a higher semantic similarity 🤔

Reply:
I was expecting it too; on one hand, German is quite strongly cased (meaning that nouns are always with a Capital Letter), on the other hand this is a general cased model. I was unable to find an uncased one, and I am reluctant to try to train one myself...

Reply:
@yosukehigashi in case we wanted, we could wrap the "cased" in a .lower() wrapper, but then of course the test would be meaningless... I am not really sure, to be honest

Reply:
Got it! It sounds like it's not _that_ surprising, so I think it's ok for now :)